src/jit/morph.cpp

```diff
@@ -10969,7 +10961,7 @@ GenTreePtr          Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* ma
             {
                 cns2 = op2;
                 /* Check for "expr relop 1" */
-                if (cns2->gtIntCon.gtIconVal == +1)
+                if (cns2->IsIOne())
```

Comment:
This condition allows long const one and not a correctness issue here because of the context around this condition.  We should avoid these.   Plus is it is unnecessarily checking whether cns2 is GT_CNS_INT, though it is already checked above.